releasenotes/notes/fix-apply-layout-duplicate-negative-indices-cf5517921fe52706.yaml

```
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    Fixed :meth:`~.SparsePauliOp.apply_layout` and :meth:`~.Pauli.apply_layout`
+    to raise ``QiskitError`` if duplicate indices or negative indices are provided
```

```suggestion
    Fixed :meth:`.SparsePauliOp.apply_layout` and :meth:`.Pauli.apply_layout`
    to raise :exc:`QiskitError` if duplicate indices or negative indices are provided
```

The leading `~` strips all but the rightmost part of the path, so `` :meth:`~.SparsePauliOp.apply_layout` `` and `` :meth:`~.Pauli.apply_layout` `` would both appear in the text as `apply_layout()` - dropping the `~` means we'll see the class as well.

---

Thank you for the information about `~`. I didn't know the effect and use it always.

---

It's usually what we want for classes, so we use it a lot - it lets you use more prefix parts in the lookup to disambiguate things, e.g. `` :class`~.circuit.Instruction` `` and `` :class:`~qiskit.pulse.Instruction` `` will both display only `Instruction`, but will refer to different objects (and for those two, it's clear from context which is meant).  It mostly only starts to be confusing when where talking about methods from outside of the class description itself (like here).  Sometimes we do say things like

> The `` :meth:`~.SparsePauliOp.to_matrix` `` and `` :meth:`~.SparsePauliMatrix.from_operator` `` methods of `` :class:`.SparsePauliOp` `` [...]

and use the `~` with the methods because we're mentioning the class separately.